cmd/mt-kafka-mdm-report-out-of-order/main.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"context"
+	"math/rand"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	inKafkaMdm "github.com/grafana/metrictank/input/kafkamdm"
+	"github.com/grafana/metrictank/logger"
+	log "github.com/sirupsen/logrus"
+)
+
+func configureLogging() {
+	formatter := &logger.TextFormatter{}
+	formatter.TimestampFormat = "2006-01-02 15:04:05.000"
+	log.SetFormatter(formatter)
+	log.SetLevel(log.InfoLevel)
+}
+
+func filter(tracker Tracker, prefix string, substr string) {
+	for key, track := range tracker {
+		if prefix != "" && !strings.HasPrefix(track.Name, prefix) {
+			delete(tracker, key)
+		}
+		if substr != "" && !strings.Contains(track.Name, substr) {
+			delete(tracker, key)
+		}
+	}
+}
+
+func aggregateAndLog(tracker Tracker, groupByName bool, groupByTag string) {
+	count := 0
+	outOfOrderCount := 0
+	duplicateCount := 0
+	for _, track := range tracker {
+		count += track.Count
+		outOfOrderCount += track.OutOfOrderCount
+		duplicateCount += track.DuplicateCount
+	}
+
+	var aggregatedByName map[string]Aggregate
+	if groupByName {
+		aggregatedByName = aggregateByName(tracker)
+	}
+
+	var aggregatedByTag map[string]Aggregate
+	if groupByTag != "" {
+		aggregatedByTag = aggregateByTag(tracker, groupByTag)
+	}
+
+	log.Infof("total metric points count=%d", count)
+
+	log.Infof("total out-of-order metric points count=%d", outOfOrderCount)
+	if groupByName && outOfOrderCount > 0 {
+		log.Info("out-of-order metric points grouped by name:")
+		for name, aggregate := range aggregatedByName {
+			if aggregate.OutOfOrderCount > 0 {
+				log.Infof("out-of-order metric points for name=%q count=%d percentGroup=%f percentClass=%f percentTotal=%f", name, aggregate.OutOfOrderCount, float64(aggregate.OutOfOrderCount)/float64(aggregate.Count)*100, float64(aggregate.OutOfOrderCount)/float64(outOfOrderCount)*100, float64(aggregate.OutOfOrderCount)/float64(count)*100)
```

Comment:
Worth making a small function to form % from 2 counts to clean this up a bit? then these all can be something like `asPercent(aggregate.OutOfOrderCount, aggregate.Count)`